src/QMCHamiltonians/ForceCeperley.cpp

```diff
@@ -36,11 +36,30 @@ ForceCeperley::ForceCeperley(ParticleSet& ions, ParticleSet& elns)
   m_exp   = 2;
   N_basis = 4;
   forces  = 0.0;
-  forces_ShortRange.resize(Nnuc);
-  forces_ShortRange = 0.0;
+  ions.update();
+  evaluate_IonIon();
   ///////////////////////////////////////////////////////////////
 }
 
+void ForceCeperley::evaluate_IonIon()
+{
+  forces_IonIon = 0.0;
```

Comment:
Please pass `forces_IonIon` as a function argument (reference). Avoid the same name as the class member.
Then we can mark this function const. This should give us cleaner code.

Reply:
The current way is how it's done in the ForceChiesaPBCAA. I'd rather be consistent for now, and do a complete makeover of the force estimators later, if needed. Actually ForceBase would be the most ideal place to do the ion-ion terms, but it's currently missing. What do you think?

Reply:
I don't think you can move it to ForceBase, PBC and non PBC are different.
I was not asking to move forces_IonIon out of the base.
I mean change the call from `evaluate_IonIon()` to `evaluate_IonIon(forces_IonIon)`.
define `evaluate_IonIon(ParticleSet::ParticlePos_t& forces)` instead of `evaluate_IonIon()`

Reply:
Not quite as simply of course, but combining the two: non-PBC from ForceCeperley and PBC from ForceChiesaPBCAA, because the ion-ion terms should remain the same regardless of the estimator.
I understood what your meaning, but I was trying to be consistent with the current functions in ForceChiesaPBCAA. anyway, the direction is good, so I (hopefully) made the changes you requested.